taxonomy/algolia/serializers.py

```diff
@@ -0,0 +1,152 @@
+# -*- coding: utf-8 -*-
+"""
+Serializers to convert db data to a dictionary that algolia can understand.
+
+This module depends on serializers provided by django-rest-framework.
+"""
+
+from rest_framework import serializers
+
+from ..models import Job, JobPostings, JobSkills
+
+
+class JobPostingSerializer(serializers.ModelSerializer):
+    """
+    JobPosting serializer for algolia index.
+
+    This serializer will contain all of the metadata related to the job posting.
+    """
+    class Meta:
+        model = JobPostings
+        fields = (
+            'job_id', 'median_salary', 'median_posting_duration', 'unique_postings', 'unique_companies',
+        )
+        read_only_fields = fields
+
+
+class JobSerializer(serializers.ModelSerializer):
+    """
+    Job serializer for algolia index.
+
+    This serializer will contain all of the metadata related to jobs and will also included metadata for skills and
+    courses.
+    """
+    skills = serializers.SerializerMethodField()
+    job_postings = serializers.SerializerMethodField()
+    objectID = serializers.SerializerMethodField()  # required by Algolia, e.g. "job-{external_id}"
```

Comment:
Can you please add the link to docs explaining why we need this? Also don't we need this for other serializers?

Reply:
@muhammad-ammar ref for details https://www.algolia.com/doc/api-client/methods/indexing/#the-objectid

I have also added this reference in the code.